src/sage/data_structures/stream.py

```diff
@@ -1863,10 +2469,10 @@ def get_coefficient(self, n):
             sage: [h.get_coefficient(i) for i in range(10)]
             [0, 0, 1, 6, 20, 50, 105, 196, 336, 540]
         """
-        return sum(l * self._right[n - k]
-                   for k in range(self._left._approximate_order,
-                                  n - self._right._approximate_order + 1)
-                   if (l := self._left[k]))
+        return ZZ.sum(l * self._right[n - k]
```

Comment:
Is `ZZ` always the correct ring?

Reply:
At least, it seems to work (through coercion) :-)

There was a decision at the beginning of the project that streams should not have a base ring.  The idea was that they are more of a data structure.

In any case, this question is not within the scope of this pull request.  The change here is mostly to avoid the necessity to check whether an element in the stream has a parent.

Reply:
I think I would be more comfortable with `sum(....., default=ZZ(0))`, which would seem to express this intention ("result should be an element") better